server/src/main/java/io/druid/indexing/overlord/IndexerMetadataStorageCoordinator.java

```diff
@@ -45,5 +53,11 @@
 
   public void deleteSegments(final Set<DataSegment> segments) throws IOException;
 
+  /**
+   * Get segments safe for deletion for the datasource over the interval
```

Comment:
it's unclear from the docs whether it only applies to segments contained within the interval or overlapping the interval. Let's try to be specific.